Tests/UnitTests/Plugins/GeoModel/GeoModelMuonMockupBluePrintTests.cpp

```diff
@@ -0,0 +1,236 @@
+  // the vector with the chamber nodes and the volumes
+  std::vector<std::shared_ptr<StaticBlueprintNode>> volChamberNodes;
+  std::vector<std::unique_ptr<Acts::TrackingVolume>> volChambers;
+
+  for (std::size_t sector = 0; sector < nSectors; sector++) {
+    ACTS_DEBUG("Barrel name: " << name << " sector: " << sector);
+    // Find the bounding box for the given chamber name and sector
+    auto it_first =
+        std::ranges::find_if(boundingBoxes, [&name, &sector](const auto& pair) {
+          return pair.first->name().find(name) != std::string::npos &&
+                 pair.first->name().find("_1_" + std::to_string(sector + 1)) !=
+                     std::string::npos;
+        });
+
+    // Check if the bounding box was found
+    BOOST_CHECK(it_first != boundingBoxes.end());
+
+    auto it_second = std::find_if(
+        it_first, boundingBoxes.end(), [&name, &sector](const auto& pair) {
+          return pair.first->name().find(name) != std::string::npos &&
+                 pair.first->name().find("_1_" + std::to_string(sector + 1)) !=
+                     std::string::npos;
+        });
+    // Check if the second bounding box was found
+    BOOST_CHECK(it_second != boundingBoxes.end());
+
+    auto vol1 = it_first->second;
+    auto vol2 = it_second->second;
+    Vector3 center = (vol1->center() + vol2->center()) / 2;
+    Acts::Transform3 volTrf = vol1->transform().rotation() *
+                              Acts::Transform3(Acts::Translation3(center));
+    auto volumeBounds = vol1->volumeBounds().values();
+    double ymin = std::min(vol1->center().y(), vol2->center().y());
+    double ymax = std::max(vol1->center().y(), vol2->center().y());
+
+    double yup = ymax + volumeBounds[1];
+    double ylow = ymin - volumeBounds[1];
+
+    std::shared_ptr<Acts::CuboidVolumeBounds> chamberBounds =
+        std::make_shared<Acts::CuboidVolumeBounds>(
+            volumeBounds[0], (yup - ylow), volumeBounds[2]);
+
+    std::unique_ptr<Acts::TrackingVolume> chamberVolume =
+        std::make_unique<Acts::TrackingVolume>(
+            volTrf, chamberBounds,
+            "Chamber_" + name + "_" + std::to_string(sector + 1));
+
+    // add the multilayer volumes
+    chamberVolume->addVolume(std::make_unique<TrackingVolume>(
+        *vol1, "Chamber_" + name + "_" + std::to_string(sector + 1) + "_MDT1"));
+    chamberVolume->addVolume(std::make_unique<TrackingVolume>(
+        *vol2, "Chamber_" + name + "_" + std::to_string(sector + 1) + "_MDT2"));
+
+    // Now also add the RPC plane surfaces to the chamber volume
+    for (const auto& surface : sensitiveSurfaces) {
+      auto sname = std::get<0>(surface)->databaseEntryName();
+
+      // Skip MDT surfaces and surfaces not in the current sector
+      if (sname.find("MDT") != std::string::npos ||
+          sname.find("_1_" + std::to_string(sector + 1)) == std::string::npos) {
+        continue;
+      }
+
+      chamberVolume->addSurface(std::get<1>(surface));
```

Comment:
The surfaces go directly into the `chamberVolume`, and not into the corresponding `MDT` volume?